blocks/site-hierarchy-content-block/sources/site-service-hierarchy.js

```diff
@@ -11,17 +11,11 @@ export default {
 		sectionId: "text",
 	},
 	transform: (data, query) => {
-		let idMatch = false;
-		if (query.sectionId) {
-			idMatch = data._id !== query.sectionId;
+		if (query?.sectionId === data._id) {
```

Comment:
This also needs to check if the `sectionId` was queried in the first place. This change breaks the section nav and footer since using this source for just a nav hierarchy always 404's.

Using the same logic as v2 will probably work:
```
if (query.sectionId && query.sectionId !== data._id) {
				const error = new Error("Not found");
				error.statusCode = 404;
				throw error;
}

return data;
```

Reply:
`query?.sectionId` is that check?

Reply:
oh, nm, I see what you are saying...